docs/hello_nextflow/03_hello_containers.md

```diff
@@ -1,82 +1,442 @@
+
+![Seqera Containers](img/seqera-containers-2.png)
+
+If this is the first time a community container has been built for this package, it may take a few minutes to complete.
+Click to copy the URI (e.g. `community.wave.seqera.io/library/pip_quote:ae07804021465ee9`) of the container image that was created for you.
+
+### 2.3. Use the container image
 
-TODO
+You can now use the container image to run the `quote` command and get a random saying from Grace Hopper.
 
-### 2.1. [step 1]
+```bash
+docker run --rm community.wave.seqera.io/library/pip_quote:ae07804021465ee9 quote "Grace Hopper"
+```
+
+Output:
+
+```console title="Output"
+Humans are allergic to change. They love to say, 'We've always done it
+this way.' I try to fight that. That's why I have a clock on my wall
+that runs counter-clockwise.
+```
 
-TODO
+### 2.4. STRETCH GOAL: Build the container image yourself
 
-### 2.2. [step 2]
+Go back to the Seqera Containers website and click on the "Build Details" button.
+There you can see the details of how our container image was built.
+You can view the conda environment file and the `Dockerfile` together make up the recipe for the container image.
+
+```conda.yml
+channels:
+- conda-forge
+- bioconda
+dependencies:
+- pip
+- pip:
+  - quote==3.0.0
+```
+
+```Dockerfile
+FROM mambaorg/micromamba:1.5.10-noble
+COPY --chown=$MAMBA_USER:$MAMBA_USER conda.yml /tmp/conda.yml
+RUN micromamba install -y -n base -f /tmp/conda.yml \
+    && micromamba install -y -n base conda-forge::procps-ng \
+    && micromamba env export --name base --explicit > environment.lock \
+    && echo ">> CONDA_LOCK_START" \
+    && cat environment.lock \
+    && echo "<< CONDA_LOCK_END" \
+    && micromamba clean -a -y
+USER root
+ENV PATH="$MAMBA_ROOT_PREFIX/bin:$PATH"
+```
+
+Copy the contents of these files into the stubs located in the `containers/build` directory, then run the following command to build the container image yourself.
+
+```bash
+docker build -t quote:latest containers/build
+```
+
+After it has finished building, you can run the container image you just built.
+
+```bash
+docker run --rm quote:latest quote "Margaret Oakley Dayhoff"
+```
+
+!!! Hint
+
+    Even if Seqera Containers doesn't manage to successfully build a container image for you, the `Dockerfile` and `conda.yml` are great starting point for a manual build.
+
+It often only takes a few additional `RUN` commands added to the Dockerfile to add the missing dependencies or system libraries.
+
+### Takeaway
+
+You know how to get a find/build a container image for any pip/conda installable tool using Seqera Containers.
+
+### What's next?
+
+Learn how to use containers in Nextflow.
+
+---
+
+## 3. Use containers in Nextflow
+
+Nextflow has built-in support for running processes inside containers.
+This means that you can use any container image you like to run your processes, and Nextflow will take care of pulling the image, mounting the data, and running the process inside it.
```

Comment:
make clear / reiterate this means you can run tools not installed in your compute env

Reply:
Done